### DISCUSSION TITLE
Estimate relative camera poses on rigid rig. Can Colmap do this?

### DISCUSSION THREAD

**Discussion by Giffffff** [colmap/colmap#2586] [Q&A]
Hi,

I got some data captured using multiple (8), time-synchronised cameras, mounted on a rigid rig. The rig's calibration is inaccurate and I don't have access to the checkerboard videos nor the rig itself, so I would like to try and use Colmap to correct (at least) the extrinsics of the cameras from the data in the images.

Given that the scene is a non-static one, I can only reliably match features between images taken at the same time instance. I do not care about the motion of the overall rig or of the objects in the scene over time, what I really care about is only the relative position of 7 cameras with respect to one, primary, camera.

Is this something that I can achieve using Colmap? Do I need to use the rig configuration in https://github.com/colmap/colmap/blob/a692b6813ce76c3bffc60db067435bc80c71488f/src/colmap/exe/sfm.cc#L536 (it does not seem to be extensively documented). Any tips/guidance would be greatly appreciated.

Thanks

**Comment by QMonteCarlo** [comment]
I've used COLMAP to calibrate multi-camera rig, you can just go. However, this is not provided in the GUI as far as I know, you may need to do it using the  commandline. COLMAP has nice command prompt, you may find it easy to finish. First you need to store the correspond images of each camera in seperate directories and make sure they have same index and number of images so it's easy to find correspondence. Then do the sfm process in gui, save the result (txt or bin), then load it and do rig calibration using command options.

**Reply by grafpez** [reply]
Hi QMonteCarlo, I'm trying to reconstruct from 3D camera with known IPD. What's your suggestion to use as parameters for rig_configurator? I'm a nOOb needing some help ;-)

**Comment by QMonteCarlo** [comment]
sorry I don't get your abbr. please let me known what is your 3d camera,
RGBD or stereo? what is IPD? Are you doing all these with colmap?

grafpez ***@***.***> 于2025年7月23日周三 09:21写道：
